File: clj-library/group_by.py
from clj.seqs import group_by

sequence = range(1, 100)

print(group_by(lambda x: x % 3, sequence))

groups = group_by(lambda x: x % 3, sequence)

keys = sorted(groups.keys())

for key in keys:
    print(key, groups[key])

print(3 * "\n")

colors = ["red", "blue", "green", "yellow", "cyan", "magenta", "black", "white"]

groups = group_by(lambda color: len(color), colors)
print(groups)

File: clj-library/group_by.clj
(def s (range 1 100))

(def groups (group-by #(mod % 3) s))

(def k (sort (keys groups)))

(doseq [key k]
    (println key (get groups key)))

(println "\n\n\n")

(def colors ["red", "blue", "green", "yellow", "cyan", "magenta", "black", "white"])

(def groups (group-by #(count %) colors))

(println groups)
